### PR TITLE
Update SnapGeometryEdits

### PR DESCRIPTION
**Pull request by Luiy0**
## Description
This update fixes a casting issue in the [sample](https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/194) given snap sources can now be graphics overlays. Additionally, the UI has been reworked to reduce code and updated to include a section to enable snapping to the graphics overlay. 

Notes:
- The upper part of the `SnapSettingsScreen` is hidden underneath the scaffold `topBar` with the addition of the graphics overlay UI section. A workaround was to move the `SampleTopAppBar` to within the Scaffold contents so that the `SnapSettingsScreen` appears above the sample title. 
- Readme and metadata to be updated following design changes.

## Links and Data
[Update Design PR](https://devtopia.esri.com/runtime/common-samples/pull/3923)

## What To Review
-  Review the code to make sure it is easy to follow like other samples on Android
- `README.md` and `README.metadata.json` files

## How to Test
Run the sample on the sample viewer or the repo.

### CODE REVIEW

File: snap-geometry-edits/src/main/java/com/esri/arcgismaps/sample/snapgeometryedits/screens/SnapSettingsScreen.kt [hunk 1]
```diff
@@ -126,117 +127,102 @@ fun SnapSettings(
...
-                                    onSnapSourceChanged(true, index)
-                                }
-                            }
-                        }
-                    ) {
-                        Text(text = "Enable All Sources")
+                SetupSnapSourceUI(snapSourceList, isSnapSourceEnabled, onSnapSourceChanged, GeometryType.Point)
+                SetupSnapSourceUI(snapSourceList, isSnapSourceEnabled, onSnapSourceChanged, GeometryType.Polyline)
+                SetupSnapSourceUI(snapSourceList, isSnapSourceEnabled, onSnapSourceChanged, null)
+            }
```
Comment (fpearson1111) [inline snap-geometry-edits/src/main/java/com/esri/arcgismaps/sample/snapgeometryedits/screens/SnapSettingsScreen.kt:133]: Nice :+1:

**Review by fpearson1111 — APPROVED** [review]
LGTM (haven't done a super in-depth review though)

**Comment by Luiy0** [comment]
Hi @shubham7109, can you please review? Thanks.

File: snap-geometry-edits/src/main/java/com/esri/arcgismaps/sample/snapgeometryedits/screens/SnapSettingsScreen.kt [hunk 2]
```diff
@@ -126,117 +127,102 @@ fun SnapSettings(
...
+        }
+    }
+}
+
+@Composable
+fun SetupSnapSourceUI(
```
Comment (hud10837) [inline snap-geometry-edits/src/main/java/com/esri/arcgismaps/sample/snapgeometryedits/screens/SnapSettingsScreen.kt:139]: Better to make this a noun, ie. `SnapSourceUI`

**Review by hud10837 — APPROVED** [review]
Changes LGTM. One naming change, but no need to re-review.

There is one preexisting bug in this sample that I found, which your changes haven't caused. If I start creating a geometry but never tap on the map, then want to cancel, pressing the delete icon doesn't work, I can only stop by pressing the save icon. That's a bit unintuitive but not a huge deal.

**Comment by Luiy0** [comment]
Thanks for the quick review @hud10837. I made a quick fix to the delete button as per your observation. We should be ready to merge now.

**Pull request merged**